src/utils/frame.js

```diff
@@ -14,163 +14,113 @@ function getFrameUrl(frame) {
 }
 
 function isEmber(frame) {
-  return getFrameUrl(frame).match(/ember/i);
+  return getFrameUrl(frame).match(/ember/i) && "Ember";
 }
 
 function isChoo(frame) {
-  return getFrameUrl(frame).match(/choo/i);
+  return getFrameUrl(frame).match(/choo/i) && "Choo";
 }
 
 function isVueJS(frame) {
-  return getFrameUrl(frame).match(/vue\.js/i);
+  return getFrameUrl(frame).match(/vue\.js/i) && "VueJS";
 }
 
 function isRxJs(frame) {
-  return getFrameUrl(frame).match(/rxjs/i);
+  return getFrameUrl(frame).match(/rxjs/i) && "RxJS";
 }
 
 function isAngular(frame) {
-  return getFrameUrl(frame).match(/angular/i);
+  return getFrameUrl(frame).match(/angular/i) && "Angular";
 }
 
 function isRedux(frame) {
-  return getFrameUrl(frame).match(/redux/i);
+  return getFrameUrl(frame).match(/redux/i) && "Redux";
 }
 
 function isDojo(frame) {
-  return getFrameUrl(frame).match(/dojo/i);
+  return getFrameUrl(frame).match(/dojo/i) && "Dojo";
 }
 
 function isPreact(frame) {
-  return getFrameUrl(frame).match(/preact/i);
+  return getFrameUrl(frame).match(/preact/i) && "Preact";
+}
+
+function isMarko(frame) {
+  return getFrameUrl(frame).match(/marko/i) && "Marko";
 }
 
 export function getLibraryFromUrl(frame: Frame) {
   // @TODO each of these fns calls getFrameUrl, just call it once
   // (assuming there's not more complex logic to identify a lib)
 
-  if (isBackbone(frame)) {
-    return "Backbone";
-  }
-
-  if (isJQuery(frame)) {
-    return "jQuery";
-  }
-
-  // Needs to remain before "react", otherwise "react" can also match "preact"
-  if (isPreact(frame)) {
-    return "Preact";
-  }
-
-  if (isReact(frame)) {
-    return "React";
-  }
-
-  if (isWebpack(frame)) {
-    return "Webpack";
-  }
-
-  if (isNodeInternal(frame)) {
-    return "Node";
-  }
-
-  if (isExpress(frame)) {
-    return "Express";
-  }
-
-  if (isChoo(frame)) {
-    return "Choo";
-  }
-
-  if (isPug(frame)) {
-    return "Pug";
-  }
-
-  if (isExtJs(frame)) {
-    return "ExtJS";
-  }
-
-  if (isUnderscore(frame)) {
-    return "Underscore";
-  }
-
-  if (isLodash(frame)) {
-    return "Lodash";
-  }
-
-  if (isEmber(frame)) {
-    return "Ember";
-  }
-
-  if (isVueJS(frame)) {
-    return "VueJS";
-  }
-
-  if (isRxJs(frame)) {
-    return "RxJS";
-  }
-
-  if (isAngular(frame)) {
-    return "Angular";
-  }
-
-  if (isRedux(frame)) {
-    return "Redux";
-  }
-
-  if (isDojo(frame)) {
-    return "Dojo";
-  }
-
-  if (isImmutable(frame)) {
-    return "Immutable";
-  }
+  return (
+    isBackbone(frame) ||
+    isJQuery(frame) ||
+    isPreact(frame) ||
+    isReact(frame) ||
+    isWebpack(frame) ||
+    isNodeInternal(frame) ||
+    isExpress(frame) ||
+    isChoo(frame) ||
+    isPug(frame) ||
+    isExtJs(frame) ||
+    isUnderscore(frame) ||
+    isLodash(frame) ||
+    isEmber(frame) ||
+    isVueJS(frame) ||
+    isRxJs(frame) ||
+    isAngular(frame) ||
+    isRedux(frame) ||
+    isDojo(frame) ||
+    isImmutable(frame) ||
+    isMarko(frame)
+  );
```

Comment:
thanks... but i'm not sure if we want to clean up the code yet. I'm not sure what API i want yet :)

Reply:
The function had 20 if conditions, giving me linting error of function complexity 20. So, I was forced to do the cleanup. I can revert back. But then I have to disable this particular linting error. Do you have any other suggestion?

Reply:
I think a solution that would be nicer might look something like this. I wouldn't say it needs to be done here in this PR though.  

```js
const libraryMap = {
  react: {
    label: "React",
    pattern: /react/i,
  },
  ...
};

export function getLibraryFromUrl(frame: Frame) {
  const frameUrl = getFrameUrl(frame);
  return Object
    .values(libraryMap)
    .find(({ pattern }) => frameUrl.match(pattern))
    .label; 
}
```

Reply:
haha - okay :)

i think @wldcordeiro has a nice refactor... lets do it 😛

Reply:
Great. Even I think a map is better. I will refactor this.

Reply:
@jasonLaster Refactoring done.